### ISSUE TITLE
Warn on `NaN` loss

### ISSUE THREAD

**Issue by KronosTheLate** [FluxML/Flux.jl#1981]
Would it not be convenient, particularily for users with less experienced with artificial neural networks, to print a warning when trying to update parameters with a loss of NaN? This could save a lot of wasted resources.

**Comment by mcabbott** [comment]
Yes probably. 

One option would be to add this to Zygote, so that any gradient call will alert you. Perhaps by adding a check to `Zygote.sensitivity`.

Another would be to add this only to the higher-level `Flux.train!`. In this case perhaps it should simply throw an error: If the loss is NaN, then the gradient will be too, and so `update!` will overwrite all weights with NaN --- stopping before that is probably more useful.
